feature/experimental/p4rt/otg_tests/traceroute_packetout_test/packetout_test.go

```diff
@@ -57,74 +76,68 @@ func sendPackets(t *testing.T, client *p4rt_client.P4RTClient, packets []*p4v1.P
 	}
 }
 
-// testPacketOut sends out PacketOut with payload on p4rt leader or
-// follower client, then verify DUT interface statistics
+// testPacketOut sends out PacketOut with payload on p4rt client,
+// then verify DUT interface statistics
 func testPacketOut(ctx context.Context, t *testing.T, args *testArgs) {
-	leader := args.leader
-	desc := "PacketOut from Primary Controller"
 	ttl := 2
-	//for ipv4
-	t.Run(desc+" ipv4 ", func(t *testing.T) {
-		// Check initial packet counters
-		port := sortPorts(args.ate.Ports())[0].ID()
-		t.Logf("Sending ipv4 pakcets with ttl %d", ttl)
-		counter0 := gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Port(port).Counters().InFrames().State())
-		t.Logf("Initial number of packets: %d", counter0)
-
-		packetCounter := 100
-		packets, err := args.packetIO.GetPacketOut(args.srcMAC, args.dstMAC, portId, true, uint8(ttl), packetCounter)
-		if err != nil {
-			t.Fatalf("GetPacketOut returned unexpected error: %v", err)
-		}
-		t.Logf("Sending packets now")
 
-		sendPackets(t, leader, packets)
+	counter0p1 := gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Port("port1").Counters().InFrames().State())
+	t.Logf("Initial number of packets on ATE port %s: %d", "port1", counter0p1)
+	counter0p2 := gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Port("port2").Counters().InFrames().State())
+	t.Logf("Initial number of packets on ATE port %s: %d", "port2", counter0p2)
 
-		// Wait for ate stats to be populated
-		time.Sleep(60 * time.Second)
-
-		// Check packet counters after packet out
-		counter1 := gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Port(port).Counters().InFrames().State())
-		t.Logf("Final number of packets: %d", counter1)
-
-		// Verify InPkts stats to check P4RT stream
-		t.Logf("Received %v packets on ATE port %s", counter1-counter0, port)
+	packets, err := args.packetIO.GetPacketOut(args.srcMAC, args.dstMAC, args.useIpv6, uint8(ttl), 100, args.metadata)
```

Comment:
GetPacketOut() takes a `isIpv4` bool as the third argument, however here you supply `useIpv6` instead. I.e. it appears the logic here was accidentally inversed?

Reply:
Right! thanks @LimeHat. Changed to useIpv4.